### PR TITLE
feat(ci): docker and docker compose support

### PR DESCRIPTION
**Pull request by debendraoli**
This pr adds docker build defination for goreleaser builds.

Darwin builds release are removed due to compatibility issue simplifying the release process.

Also adds support for docker compose deployment with ease.

### CODE REVIEW

**Comment by bcsainju** [comment]
I couldn't build and i am getting errors . any other configuration required ?? Also please fix the github checks.

File: .goreleaser.yaml [hunk 1]
```diff
@@ -87,21 +47,51 @@ builds:
...
+    use: buildx
+    build_flag_templates:
+      - "--label=org.opencontainers.image.created={{.Date}}"
+      - "--label=org.opencontainers.image.title={{.ProjectName}}"
+      - "--label=org.opencontainers.image.revision={{.FullCommit}}"
+      - "--label=org.opencontainers.image.version={{.Version}}"
+      - "--platform=linux/amd64"
+  - image_templates:
+      - iconcommunity/{{.ProjectName}}:{{ .Tag }}-arm64v8
+    dockerfile: goreleaser.dockerfile
```
Comment (bcsainju) [inline .goreleaser.yaml:63]: Why aren't we using the existing dockerfile and using scratch ? using debian significantly increases the docker image size.

Comment (debendraoli) [reply .goreleaser.yaml:63]: Because we're using CGO and that makes it system dependent.

And we've already built the binary just need to copy it.

Comment (bcsainju) [reply .goreleaser.yaml:63]: can you give me a breakdown on the docker size ?

Comment (debendraoli) [reply .goreleaser.yaml:63]: AMD64: 72.02MB
ARm64: 300.25MB

Comment (debendraoli) [reply .goreleaser.yaml:63]: > Why aren't we using the existing dockerfile and using scratch ? using debian significantly increases the docker image size.

more info here:

https://goreleaser.com/errors/docker-build/

I had to struggle for couple of days for this mistake when I was using scratch image.

Comment (bcsainju) [reply .goreleaser.yaml:63]: I meant before and after docker image as we already had docker image based on scratch. Based on the increment of the image size, we can discuss further.

Comment (debendraoli) [reply .goreleaser.yaml:65]: Never used it, thats not based on scratch but alpine one unlike prod build.

Comment (debendraoli) [reply .goreleaser.yaml:65]: That dockerfile can be safely purged without consequences.

Comment (bcsainju) [reply .goreleaser.yaml:65]: that is the basis for all e2e tests and cannot be purged but if we are to make docker for relay, the best one would be to utilize similar pattern.

**Pull request closed without merging**